Lib/traceback.py

```diff
@@ -1732,6 +1710,26 @@ def _compute_suggestion_error(exc_value, tb, wrong_name):
     return suggestion
 
 
+def _compute_suggestion_message(exc_type, exc_value, exc_traceback) -> str | None:
+    if issubclass(exc_type, ModuleNotFoundError) and \
+        sys.flags.no_site and \
+        getattr(exc_value, "name", None) not in sys.stdlib_module_names:
+            return ("Site initialization is disabled, did you forget to "
+                "add the site-packages directory to sys.path?")
+    if issubclass(exc_type, (ImportError, NameError, AttributeError)):
+        name_attribute = "name_from" if issubclass(exc_type, ImportError) else "name"
+        wrong_name = getattr(exc_value, name_attribute, None)
+        if wrong_name:
+            suggestion_error = _compute_suggestion_error(exc_value, exc_traceback, wrong_name)
+            suggestion_text = f"Did you mean: '{suggestion_error}'?" if suggestion_error else None
+            if issubclass(exc_type, NameError) and wrong_name in sys.stdlib_module_names:
+                if suggestion_error:
+                    return f"{suggestion_text} Or did you forget to import '{wrong_name}'?"
+                return f"Did you forget to import '{wrong_name}'?"
+            return suggestion_text
```

Comment:
I think the following could be a bit clearer:

```suggestion
            other_name = _compute_suggestion_error(
                exc_value, exc_traceback, wrong_name
            )
            maybe_builtin_import = (
                issubclass(exc_type, NameError)
                and wrong_name in sys.stdlib_module_names
            )
            if not other_name:
                if maybe_builtin_import:
                    return f"Did you forget to import '{wrong_name}'?"
                return None
            text = f"Did you mean: '{other_name}'?"
            if maybe_builtin_import:
                return f"{text} Or did you forget to import '{wrong_name}'?"
            return text
```